### ISSUE TITLE
Refactor configuration options

### ISSUE THREAD

**Issue by gegoune** [NeogitOrg/neogit#227]
**Is your feature request related to a problem? Please describe.**
Options aren't 'in the same direction' at the moment. Would be nice to remove `disable_` strings from them and switch bools around.

```lua
{
  disable_signs = false,
  disable_hint = false,
  disable_context_highlighting = false,
  disable_commit_confirmation = false,
  auto_refresh = true,
  disable_builtin_notifications = false,
}
```

This is subset of current options. Please note how `auto_refresh` is 'positive' while all others are 'negative' (`disable_`).

**Describe the solution you'd like**
 It would probably make sense to group related options into subtable and turn them around, so instead of `disable_signs = false` default would be `signs = true`, but also would be nested into, let's say, `display` subtable (`display.signs = true`).

**Describe alternatives you've considered**
Leave things as they are but setup table will eventually become polluted.

**Additional context**
https://github.com/TimUntersberger/neogit/pull/225#discussion_r724809237

**Comment by CKolkey** [comment]
@treatybreaker this could be something nice to do, yea?

**Comment by treatybreaker** [comment]
> @treatybreaker this could be something nice to do, yea?

Yeah, this would be solid. Not sure about the nested scope though. I don't think there's a ton of value in nesting the scope unless there's other values within that scope or we have repeated similar configs like with the `kind` stuff for `commit_editor`, `commit_view`, etc.

**Comment by CKolkey** [comment]
Dealers choice, I'm mostly interested in flipping the negative (`disable_commit_confirm = true`) into the positive `commit_confirm = true`

**Comment by D00mch** [comment]
The pull request with the fix is here https://github.com/NeogitOrg/neogit/pull/1556
